cnf-certification-test/accesscontrol/securitycontextcontainer/securitycontextcontainer.go

```diff
@@ -0,0 +1,267 @@
+)
+
+func GetContainerSCC(cut *v1.Container, containerSCC ContainerSCC) ContainerSCC {
+	containerSCC.HostPorts = false
+	for _, aPort := range cut.Ports {
+		if aPort.HostPort != 0 {
+			containerSCC.HostPorts = true
+			break
+		}
+	}
+	containerSCC = updateCapabilities(cut, containerSCC)
+	if cut.SecurityContext != nil && cut.SecurityContext.AllowPrivilegeEscalation != nil {
+		logrus.Info("PrivilegeEscalation is true")
+		containerSCC.PrivilegeEscalation = true
+	}
+	containerSCC.PrivilegedContainer = false
+	if cut.SecurityContext != nil && cut.SecurityContext.Privileged != nil {
+		containerSCC.PrivilegedContainer = *(cut.SecurityContext.Privileged)
+	}
+	if cut.SecurityContext != nil && cut.SecurityContext.RunAsUser != nil {
+		logrus.Info("RunAsUser is ", cut.SecurityContext.RunAsUser)
+		containerSCC.RunAsUser = true
+	}
+	containerSCC.ReadOnlyRootFilesystem = false
+	if cut.SecurityContext != nil && cut.SecurityContext.ReadOnlyRootFilesystem != nil {
+		containerSCC.ReadOnlyRootFilesystem = *cut.SecurityContext.ReadOnlyRootFilesystem
+	}
+	containerSCC.RunAsNonRoot = false
+	if cut.SecurityContext != nil && cut.SecurityContext.RunAsNonRoot != nil {
+		logrus.Info("RunAsNonRoot is ", *(cut.SecurityContext.RunAsNonRoot))
+		containerSCC.RunAsNonRoot = *cut.SecurityContext.RunAsNonRoot
+	}
+	if cut.SecurityContext != nil && cut.SecurityContext.SELinuxOptions != nil {
+		containerSCC.SeLinuxContext = true
+	}
+	return containerSCC
+}
+
+func updateCapabilities(cut *v1.Container, containerSCC ContainerSCC) ContainerSCC {
+	containerSCC.HaveDropCapabilities = "notHaveRequiredDrop"
+	if cut.SecurityContext != nil && cut.SecurityContext.Capabilities != nil {
+		var sliceDropCapabilities []string
+		for _, ncc := range cut.SecurityContext.Capabilities.Drop {
+			sliceDropCapabilities = append(sliceDropCapabilities, string(ncc))
+		}
+		logrus.Info("cut.SecurityContext.Capabilities.Drop", cut.SecurityContext.Capabilities.Drop)
+		sort.Strings(sliceDropCapabilities)
+		logrus.Info("sliceDropCapabilities", sliceDropCapabilities)
+
+		sort.Strings(requiredDropCapabilities)
+		if reflect.DeepEqual(sliceDropCapabilities, requiredDropCapabilities) || reflect.DeepEqual(sliceDropCapabilities, dropAll) {
+			containerSCC.HaveDropCapabilities = haveRequiredDrop
+		}
+
+		if checkContainCateegory(cut.SecurityContext.Capabilities.Add, category2AddCapabilities) {
+			logrus.Info("category is category2")
+			containerSCC.Capabilities = "category2"
+		} else {
+			if checkContainCateegory(cut.SecurityContext.Capabilities.Add, category3AddCapabilities) {
+				containerSCC.Capabilities = "category3"
+			} else {
+				if len(cut.SecurityContext.Capabilities.Add) > 0 {
+					containerSCC.Capabilities = "category5"
+				} else {
+					logrus.Info("category is category1")
+					containerSCC.Capabilities = category1
+				}
+			}
+		}
+	} else {
+		containerSCC.Capabilities = category1
+	}
+	return containerSCC
+}
+
+func AllVolumeAllowed(volumes []v1.Volume) bool {
```

Comment:
We should probably have a unit test for this function.  It can be added after this merges.